File: Algorithms/basic_algorithm/SelectionSort.java
package basic_algorithm;

public class SelectionSort {
	
//	实现选择排序
	public static void selectionSort(int[] arr) {
		if (arr == null || arr.length < 2) {
			return;
		}
		for (int i = 0; i < arr.length - 1; i++) {
//			规定开始的位置
			int minIndex =i;
//			minIndex 记录最小数的下标
			for (int j = i + 1; j < arr.length; j++) {
				minIndex = arr[j] < arr[minIndex] ? j : minIndex;
			}
			swap(arr, i, minIndex);
		}
	}
	
	public static void swap(int[] arr, int i, int j) {
		int tmp = arr[i];
		arr[i] = arr[j];
		arr[j] = tmp;
	}
}

File: Algorithms/basic_algorithm/SortAlgorithmsbyJavaScript/SelectionSort.js
/**
 * @author SkylineBin
 * @time 2018-10-6
 * @function use javascript to create SelectionSort
 * 
 */

module.exports = function SelectionSort(arrays){
    if (arrays == null || arrays.length < 2) {
        return;
    }
    for (let index = 0; index < arrays.length - 1; index++) {
        let minIndex = index;
        for (let j = index + 1; j < arrays.length; j++) {
            minIndex = arrays[j] < arrays[minIndex] ? j : minIndex;            
        }
        swap(arrays, index, minIndex);
        console.log(arrays);
    }
}

 function swap(arrays, si, sj) {
     let tempdata = arrays[si];
     arrays[si] = arrays[sj];
     arrays[sj] = tempdata;
 }